### PR TITLE
Added TOC and basic PR guidelines

### PR DESCRIPTION
**Pull request by jdmunro**
As PRs are such an important aspect of our daily development workflow, I thought we'd benefit from having a discussion here about important practices that we can use to make our PRs better.

I've added some basic information but please feel free to suggest new changes or disagree!

I've also introduced a table of contents as the document is already getting quite large. We'll need to manually keep the TOC up-to-date.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,31 @@
-#  The RNF Digital Engineering Handbook
+# The RNF Digital Engineering Handbook
 
 This handbook is intended as a guide to development for software engineers at RNF Digital.
 
+# Table of Contents
```
Comment (thekarel) [inline README.md]: If the document is getting large maybe it's time to split it into multiple files. Links can be created with Markdown and a tool like https://www.gitbook.com/ might also help readability. Example: https://raw.githubusercontent.com/reactjs/redux/master/docs/README.md

Comment (jdmunro) [inline README.md]: I see the need to split it up too. I think we're still at a pretty early stage at the minute and adding in the complexity of managing multiple files might put people off contributing.

If you look at the TOC I've added you'll see we have some work to do to restructure the content into a structure that makes sense. Let's tackle this within one document first and then we can split up later on.

Bit by bit we can build this up into a really useful resource 👍

Comment (jdmunro) [inline README.md]: We did actually start with a GitBook btw but we decided it was a little to complex for our needs at the time 📘

Comment (DaleJefferson) [inline README.md]: Hopefully we will get so many contributions that we nave to split it up 👍

Comment (javierM84) [inline README.md]: I have been checking GitBook and it is really good, I didn't see how we tried to use it but it looks like a similar tool will be needed when TOC becomes large

File: README.md [hunk 2]
```diff
@@ -31,6 +55,34 @@ const foo = (a, b) => a + b
...
+This section outlines our best practices for daily development.
+
+### Creating Pull Requests 
+
+Changes to any of our codebases are made through a pull request (PR). Ideally, pull requests would be concise and focussed. Here are some guidelines to help you create a great pull request:
+
+#### Keep it Focussed
+
+The best pull requests change only a few lines of code. Naturally, this isn't always possible, however you should aim to avoid 'kitchen sink' pull requests that make large amounts of seemingly unrelated changes across a codebase. This makes the task of reviewing pull requests much easier for other developers and will likely increase the quality of the feedback you receive.
+
```
Comment (javierM84) [inline README.md]: Maybe you can give a bad and good example

Comment (jdmunro) [inline README.md]: I gave a good example. I'd say a bad example is one with no description :)

**Pull request merged**